opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -864,12 +907,20 @@ namespace Opm {
             }
 
             ++it;
+            int exception_thrown = 0;
             if( localWellsActive() )
             {
                 for (auto& well : well_container_) {
-                    well->solveEqAndUpdateWellState(well_state_);
+                    try {
+                        well->solveEqAndUpdateWellState(well_state_, deferred_logger);
+                    } catch (...) {
+                        exception_thrown = 1;
```

Comment:
No break? Also here: try{} the loop as a whole.

Reply:
done